File: app/src/main/java/com/frank/ffmpeg/util/ScreenUtil.java
package com.frank.ffmpeg.util;

import android.content.Context;
import android.util.DisplayMetrics;
import android.view.WindowManager;

public class ScreenUtil {

    private static DisplayMetrics getDisplayMetrics(Context context) {
        WindowManager windowManager = (WindowManager) context.getApplicationContext().getSystemService(Context.WINDOW_SERVICE);
        if (windowManager == null) {
            return null;
        }
        DisplayMetrics displayMetrics = new DisplayMetrics();
        windowManager.getDefaultDisplay().getMetrics(displayMetrics);
        return displayMetrics;
    }

    public static int getScreenWidth(Context context) {
        if (context == null) {
            return 0;
        }
        DisplayMetrics displayMetrics = getDisplayMetrics(context);
        return displayMetrics != null ? displayMetrics.widthPixels : 0;
    }

    public static int getScreenHeight(Context context) {
        if (context == null) {
            return 0;
        }
        DisplayMetrics displayMetrics = getDisplayMetrics(context);
        return displayMetrics != null ? displayMetrics.heightPixels : 0;
    }

    public static int dp2px(Context context, int dpValue) {
        if (context == null) {
            return 0;
        }
        DisplayMetrics displayMetrics = getDisplayMetrics(context);
        float density = displayMetrics != null ? displayMetrics.density : 0;
        return (int) (dpValue * density + 0.5f);
    }

    public static int px2dp(Context context, int pxValue) {
        if (context == null) {
            return 0;
        }
        DisplayMetrics displayMetrics = getDisplayMetrics(context);
        float density = displayMetrics != null ? displayMetrics.density : 0;
        return (int) (pxValue / density + 0.5f);
    }

}

File: app/src/main/java/com/frank/ffmpeg/util/ScreenUtil.kt
package com.frank.ffmpeg.util

import android.content.Context
import android.util.DisplayMetrics
import android.view.WindowManager

object ScreenUtil {

    private fun getDisplayMetrics(context: Context): DisplayMetrics? {
        val windowManager = context.applicationContext.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        val displayMetrics = DisplayMetrics()
        windowManager.defaultDisplay.getMetrics(displayMetrics)
        return displayMetrics
    }

    fun getScreenWidth(context: Context?): Int {
        if (context == null) {
            return 0
        }
        val displayMetrics = getDisplayMetrics(context)
        return displayMetrics?.widthPixels ?: 0
    }

    fun getScreenHeight(context: Context?): Int {
        if (context == null) {
            return 0
        }
        val displayMetrics = getDisplayMetrics(context)
        return displayMetrics?.heightPixels ?: 0
    }

    fun dp2px(context: Context?, dpValue: Int): Int {
        if (context == null) {
            return 0
        }
        val displayMetrics = getDisplayMetrics(context)
        val density: Float = displayMetrics?.density ?: 0F
        return (dpValue * density + 0.5f).toInt()
    }

    fun px2dp(context: Context?, pxValue: Int): Int {
        if (context == null) {
            return 0
        }
        val displayMetrics = getDisplayMetrics(context)
        val density: Float = displayMetrics?.density ?: 0F
        return (pxValue / density + 0.5f).toInt()
    }

}
